tests/system/small/test_dataframe.py

```diff
@@ -3376,3 +3376,10 @@ def test_df_dot_operator_series(
         bf_result,
         pd_result,
     )
+
+
+def test_recursion_limit(scalars_df_index):
```

Comment:
Might be able to make this a unit test, but as discussed, it is good to exercise ibis here too.